### ISSUE TITLE
X# 2.19 (Workarea Tab) - Information is not displayed for some workareas

### ISSUE THREAD

**Issue by leon-ts** [X-Sharp/XSharpPublic#1414]
Let me start by saying thank you: the X# debugger tabs now have a fully implemented dark theme. Looks great! Thank you! And also, new very useful features have been added to them. Thanks again!

I discovered a problem with one of the new features, which is clearly demonstrated in the following video:
[ScreenRecorderProject1.zip](https://github.com/X-Sharp/XSharpPublic/files/13947854/ScreenRecorderProject1.zip)

The X# Workareas tab displays 12 workareas. When you place the cursor on workareas numbered 2 and 3, the right table does not display details (it is empty). For all other workareas (1, 4-12) the data is displayed. This session uses the SQL version of my software with my SqlRdd (CITSQL). If I run the same software in the DBF version (DBFCDX driver), then the problem described above is not observed on the “X# Workareas” tab. Apparently, the new feature has some kind of conflict with custom RDDs.

**Environment:**
X# 2.19.0.0
VS 2022 17.8.4

**Comment by RobertvanderHulst** [comment]
Leonid,
The code that is running here is this:

https://github.com/X-Sharp/XSharpPublic/blob/feature/Intellisense/src/VisualStudio/Debugger.Support/RTLink/RtLink.cs#L342

We are retrieving the RDD object and use reflection to read its public instance properties.
We build a name/value pair with the property name and stringified value. We limit this to "simple" types. 
Maybe you have a property that is write only?
I'll add  error handling to this code to skip properties that cannot be read.

**Comment by leon-ts** [comment]
Robert,
I took the method you pointed to and copied it into my project (with a few minor changes so I could run it inside my project). This method worked without problems and read all available properties. There were no exceptions. In addition, after I read the properties of the “TBSP_1” workarea using this method, the debugger also began to normally display the properties of this workarea in the “X# Workareas” tab. For clarity, I recorded this experiment on video:
[ScreenRecorderProject1.zip](https://github.com/X-Sharp/XSharpPublic/files/13949106/ScreenRecorderProject1.zip)
